crypto/ec/ec_pmeth.c

```diff
@@ -116,7 +121,16 @@ static int pkey_ec_sign(EVP_PKEY_CTX *ctx, unsigned char *sig, size_t *siglen,
     else
         type = NID_sha1;
 
-    ret = ECDSA_sign(type, tbs, tbslen, sig, &sltmp, ec);
+    if (ec_nid == NID_sm2) {
+#if defined(OPENSSL_NO_SM2)
+       ret = -1;
+#else
+       ret = SM2_sign(type, tbs, tbslen, sig, &sltmp, ec);
```

Comment:
Question about the ec_pmeth changes generally. You have enabled SM2_sign and SM2_verify to be called from pkey_ec_sign and pkey_ec_verify respectively, have you considered tying in SM2_encrypt and SM2_decrypt as well? Currently ec_pmeth.c does not provide pkey_ec_encrypt or pkey_ec_decrypt function implementations but the RSA equivalent in rsa_pmeth.c does provide pkey_rsa_encrypt and pkey_rsa_decrypt and SM2 seems to be providing an alternative to that. Is there a use case for SM2 encryption/decryption being added to ec_pmeth.c?

Reply:
Ok I should read the comments above better. I think @randombit answered my question already. Am I right in assuming that the reason the SM2_encrypt/SM2_decrypt were not hooked into ec_pmeth.c is because OpenSSL does not currently support ECIES which is where it would be utilised?